src/backend/distributed/operations/shard_transfer.c

```diff
@@ -402,14 +398,7 @@ citus_move_shard_placement(PG_FUNCTION_ARGS)
 	}
 
 	/* since this is move operation, we remove shards from source node after copy */
-	if (DeferShardDeleteOnMove)
-	{
-		MarkForDropColocatedShardPlacement(shardInterval, sourceNodeName, sourceNodePort);
-	}
-	else
-	{
-		DropColocatedShardPlacement(shardInterval, sourceNodeName, sourceNodePort);
-	}
+	DropColocatedShardPlacement(shardInterval, sourceNodeName, sourceNodePort);
```

Comment:
You kept the wrong function
```suggestion
	MarkForDropColocatedShardPlacement(shardInterval, sourceNodeName, sourceNodePort);
```